ksql-common/src/main/java/io/confluent/ksql/function/FunctionRegistry.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.function;
+
+import org.apache.kafka.connect.data.Schema;
+
+public interface FunctionRegistry {
+  KsqlFunction getFunction(String functionName);
+
+  boolean addFunction(KsqlFunction ksqlFunction);
```

Comment:
Have you thought about changing this interface to make it more like the aggregate function interface? I think it would be nice to have an interface that lets you supply the final function based on the argument expressions and schemas.

Reply:
This will change in the next PR i submit. You will get a different function depending on the argumentTypes